### PR TITLE
Add GCS support to TSDB config

### PR DESCRIPTION
**Pull request by pracucci**
**What this PR does**:
This PR adds GCS backend support to TSDB storage engine. I've both tested it with S3 (no regression) and GCS, and looks working as expected.

**Checklist**
- [x] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated

### CODE REVIEW

File: pkg/storage/tsdb/backend/gcs/config.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+// RegisterFlags registers the flags for TSDB GCS storage
+func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
+	f.StringVar(&cfg.BucketName, "experimental.tsdb.gcs.bucket-name", "", "GCS bucket name")
+	f.StringVar(&cfg.ServiceAccount, "experimental.tsdb.gcs.service-account", "", "JSON representing either a Google Developers Console client_credentials.json file or a Google Developers service account key file. If empty, fallback to Google default logic.")
+}
+
+// NewBucketClient creates a new GCS bucket client
+func (cfg *Config) NewBucketClient(ctx context.Context, name string, logger log.Logger) (objstore.Bucket, error) {
```
Comment (gouthamve) [inline pkg/storage/tsdb/backend/gcs/config.go]: Can we do `NewBucketClient(ctx, cfg, name, logger)`? Making it a method on the config struct is a new patter afaics.

File: pkg/storage/tsdb/backend/s3/config.go [hunk 2]
```diff
@@ -19,3 +25,16 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 	f.StringVar(&cfg.Endpoint, "experimental.tsdb.s3.endpoint", "", "S3 endpoint without schema")
 	f.BoolVar(&cfg.Insecure, "experimental.tsdb.s3.insecure", false, "If enabled, use http:// for the S3 endpoint instead of https://")
 }
+
+// NewBucketClient creates a new S3 bucket client
+func (cfg *Config) NewBucketClient(name string, logger log.Logger) (objstore.Bucket, error) {
```
Comment (gouthamve) [inline pkg/storage/tsdb/backend/s3/config.go]: Same here?

**Comment by pracucci** [comment]
@gouthamve Definitely, done!

**Comment by gouthamve** [comment]
@thorfour Could you take a look?

**Pull request merged**